ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -273,24 +476,33 @@ class JITModBuilder {
 
   void insertAOTMap(Instruction *AOT, Value *JIT, size_t BBIdx,
                     size_t InstrIdx) {
-    AOTMap[JIT] = {BBIdx, InstrIdx, AOT, CallDepth};
+    AOTMap[JIT] = {BBIdx, InstrIdx, AOT, CallStack.size() - 1};
   }
 
-  // Returns true if the given function exists on the call stack, which means
-  // this is a recursive call.
-  bool isRecursiveCall(Function *F) {
-    for (auto Tup : InlinedCalls) {
-      CallInst *CInst = get<1>(Tup);
-      if (CInst->getCalledFunction() == F) {
-        return true;
-      }
-    }
-    return false;
+  // Start outlining.
+  //
+  // This flips the "are we outlining" flag and marks the most-recent frame as
+  // the "outline base" (the frame where outlining started).
+  void startOutlining() {
+    assert(!Outlining);
+    MappableFrame *MPF = CallStack.curMappableFrame();
+    assert(MPF);
+    MPF->OutlineBase = true;
+    Outlining = true;
   }
 
-  // Determines whether we are currently outlining code, i.e. we are processing
-  // the IR of a function we do not wish to inline.
-  bool isOutlining() { return OutlineCallDepth > 0; }
+  // Check if we should stop outlining.
```

Comment:
From the description I assumed this would return a bool to tell us if we can stop outlining. I think the comment should be something like "Try to stop outlining if possible". And maybe we can also change the function name to `tryStopOutlining` to make it more obvious this isn't just a check, but changes the state.